app/controllers/chargeback_rate_controller.rb

```diff
@@ -4,89 +4,76 @@ class ChargebackRateController < ApplicationController
   after_action :cleanup_action
   after_action :set_session_data
 
+  include Mixins::GenericButtonMixin
+  include Mixins::GenericListMixin
   include Mixins::GenericSessionMixin
+  include Mixins::GenericFormMixin
+  include Mixins::GenericShowMixin
   include Mixins::BreadcrumbsMixin
 
-  CB_X_BUTTON_ALLOWED_ACTIONS = {
+  BUTTON_ALLOWED_ACTIONS = {
     'chargeback_rates_copy'   => :cb_rate_edit,
     'chargeback_rates_delete' => :cb_rates_delete,
     'chargeback_rates_edit'   => :cb_rate_edit,
     'chargeback_rates_new'    => :cb_rate_edit
   }.freeze
 
-  def x_button
-    generic_x_button(CB_X_BUTTON_ALLOWED_ACTIONS)
-  end
-
-  def x_show
-    @explorer = true
-    @record = identify_record(params[:id], ChargebackRate)
-    nodeid = x_build_node_id(@record)
-    params[:id] = "xx-#{@record.rate_type}_#{nodeid}"
-    params[:tree] = x_active_tree.to_s
-    tree_select
-    redirect_to :action => 'explorer' unless request.xml_http_request? # Ajax request means in explorer
-  end
+  def button
+    @edit = session[:edit]    # Restore @edit for adv search box
+    @refresh_div = "main_div" # Default div for button.rjs to refresh
+    action = params[:pressed]
 
-  def tree_select
-    self.x_active_tree = :cb_rates_tree
-    self.x_node = params[:id]
-    get_node_info(x_node)
-    replace_right_cell if request.xml_http_request? # Ajax request means in explorer
-  end
+    evaluate_button(action, BUTTON_ALLOWED_ACTIONS)
 
-  def explorer
-    @breadcrumbs = []
-    @explorer    = true
-    build_accordions_and_trees
+    return if performed?
 
-    @right_cell_text ||= _("All Chargeback Rates")
-    set_form_locals if @in_a_form
-    session[:changed] = false
+    if action.ends_with?("_copy", "_edit", "_new") && @flash_array.nil?
+      javascript_redirect(:action => @refresh_partial,  :id => @redirect_id, :pressed => action)
+    elsif action.ends_with?("_delete")
+      javascript_redirect(:action => 'show_list')
+    elsif @refresh_div == "main_div" && @lastaction == "show_list"
+      replace_gtl_main_div
+    else
+      super
+    end
+  end
 
-    render :layout => "application" unless request.xml_http_request?
+  def show_list
+    super
+    @center_toolbar = 'chargeback_rates'
```

Comment:
You can just simply use:
```ruby
toolbar :chargeback_rate, :chargeback_rates
```
outside the definition, as we do it on some other screens. This way you don't need to redefine the `show_list` and `show` methods.

```